File: ConflictsWeb/src/models/CommodityCategory.ts
export class CommodityCategory {
    public id: number;
    public type: string;
    public region: string;
    public unit: string;


    constructor(id: number, type: string, region: string, unit: string) {
        this.id = id;
        this.type = type;
        this.region = region;
        this.unit = unit;
    }
}

File: ConflictsServer/src/main/java/islab/project/conflictsserver/commodities/CommodityCategory.java
package islab.project.conflictsserver.commodities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Entity
@Table(name = "commodity_category")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CommodityCategory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String type;
    private String region;
    private String unit;

    @OneToMany(mappedBy = "commodityCategory")
    private Set<CommodityPrice> prices;
}
